From f6a499a18f5c5cf660390a4dfaf348fd9fe759c1 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@laptop.com>
Date: Fri, 14 Sep 2012 22:20:13 +0100
Subject: [PATCH] Work around for an apparent MinGW-w64 compiler (4.7.2-2) bug.

The public virtual inheritence of ResourceEditorDocument
 from Core::IDocument seems to be creating an invalid vtable.

It's now just public inheritence which is all that is needed.

The compiler in question was Ruben Van Boxem's 4.7.2-2.

Change-Id: I66f54ec04ce5aad847ac5bd6a4807fbd613c2d99
Reviewed-by: Daniel Teske <daniel.teske@digia.com>
---
 src/plugins/resourceeditor/resourceeditorw.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/resourceeditor/resourceeditorw.h b/src/plugins/resourceeditor/resourceeditorw.h
index cec40d3..5f1e04a 100644
--- a/src/plugins/resourceeditor/resourceeditorw.h
+++ b/src/plugins/resourceeditor/resourceeditorw.h
@@ -46,7 +46,7 @@ class ResourceEditorW;
 class QrcEditor;
 
 class ResourceEditorDocument
-  : public virtual Core::IDocument
+  : public Core::IDocument
 {
     Q_OBJECT
 
-- 
1.8.0.2

